CONTRIBUTING.md

```diff
@@ -0,0 +1,137 @@
+# Contributing
+
+Thank you for considering helping us with nrfx development!
+
+To make development clear and consistent, we have set up a few rules.
+Please get familiar with them before creating pull requests. It will help us to merge your changes faster.
+
+1. Check nrfx's [git workflow](#git-workflow).
+2. [Update](#updating-the-changelog) the CHANGELOG.md file if needed.
+3. Make sure your commit messages meet the [rules](#commit-message-rules).
+
+## Git workflow
+
+The nrfx project follows a simple git workflow. New feature branches are branched from the master
+branch on a forked repository and merged back to the master branch using pull requests. History is linear.
+
+### Working on branches
+Before merging a branch back to the master branch, a few requirements must be met.
+
+ * Make sure that [CLA (Contributor License Agreement)](https://cla-assistant.io/NordicSemiconductor/nrfx)
+ has been accepted.
+ * Make commits clear and consistent. Make sure that each commit is well-described and contains
+ one logical change (see [Commit message rules](#commit-message-rules) for more). If there are
+ more commits related to one logical change, squash them.
+ * Rebase the branch at the top of the master branch to be mergeable. Upmerges are prohibited.
+ * Avoid new commits for changes originating from reviews. If possible, use the interactive rebase
+ function to introduce these changes into the existing commits.
+
+## Updating the changelog
```

Comment:
It should be optional because it generates lot of conflicts when more pull-requests change CHANGELOG

Reply:
Regarding both your comments. In 19b2d8bab338a85b4e554f04601c8b507f838379, I took the content [from the wiki](https://github.com/NordicSemiconductor/nrfx/wiki/Contribution) without any modifications.
Should we merge the guide as it is now in the Wiki, then supress the Wiki and then apply your comments on top?
I agree, the changelog can be updated in the release branch. We have a clean history so it is easy to see the changes of the new version.